plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineSGXManifest.kt

```diff
@@ -0,0 +1,96 @@
+package com.r3.conclave.plugin.enclave.gradle.gramine
+
+import com.r3.conclave.common.internal.PluginUtils.GRAMINE_ENCLAVE_JAR
+import com.r3.conclave.common.internal.PluginUtils.PYTHON_FILE
+import com.r3.conclave.plugin.enclave.gradle.ConclaveTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.process.ExecResult
+import java.io.File
+import javax.inject.Inject
+
+open class GenerateGramineSGXManifest @Inject constructor(objects: ObjectFactory) : ConclaveTask() {
+    companion object {
+        const val GRAMINE_SGX_SIGN_EXECUTABLE = "gramine-sgx-sign"
+        const val GRAMINE_GET_TOKEN_EXECUTABLE = "gramine-sgx-get-token"
+    }
+
+    @get:InputFile
+    val privateKey: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val directManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val enclaveJar: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    @Optional
+    val pythonSourcePath: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxToken: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxSig: RegularFileProperty = objects.fileProperty()
+
+    override fun action() {
+        val manifestPath = directManifest.get().asFile.absolutePath
+        val privateKeyPath = privateKey.get().asFile.absolutePath
+        val outputSGXManifestPath = sgxManifest.asFile.get().parentFile.absolutePath
+        val enclaveDestinationJarName = "${outputSGXManifestPath}/$GRAMINE_ENCLAVE_JAR"
+        val enclaveDestinationJarFile = File(enclaveDestinationJarName)
+        checkNotNull(
+            enclaveJar.get().asFile.copyTo(
+                enclaveDestinationJarFile,
+                overwrite = true
+            )
+        ) { "Enclave jar file not copied correctly while builgetEnclaveThreadCountFromManifestding the Gramine SGX manifest" }
+        val enclaveDestinationPythonFileName = "${outputSGXManifestPath}/$PYTHON_FILE"
+        val enclaveDestinationPythonFile = File(enclaveDestinationPythonFileName)
+
+        if (pythonSourcePath.isPresent) {
+            checkNotNull(
+                pythonSourcePath.get().asFile.copyTo(
+                    enclaveDestinationPythonFile,
+                    overwrite = true
+                )
+            ) { "Python script not copied correctly while building the Gramine SGX manifest" }
```

Comment:
`copyTo` returns a boolean if the copy was a success or not, rather than throwing an IOException. This odd API choice is due to the underlying `java.io.File` API, which has these issues, and this is why I always say to use the newer `Path` API whenever possible (which was created to address these issues).

Reply:
`copyTo` seems to return a File actually. Somehow it is so immediate to use `asFile` from the `RegularFile` property, hence I used this, which worked.
Should I use 

```

            Files.copy(enclaveJar.get().asFile.toPath(), enclaveDestinationJarPath)

``` 
 What should I use instead?

Reply:
Yes I appreciate that Gradle uses `File` and so it's sometimes easier to use that over `Path`. I do that same as well sometimes, e.g.`writeText`. However, most times stick to `Path`, especially when you have deal with `File`'s poor API. https://www.baeldung.com/java-path-vs-file gives some more info on this.

With regards to sample, I think you can do `enclaveJar.get().asFile.toPath().copyTo(...`

Btw, why do u need to overwrite?

Reply:
No need, I have removed it.
I think I vaguely remember we had cases where copy did not succeed and Gradle was silent about this.

Reply:
That's exactly the issue with `File` API.